core/dtn7/src/routing/erouting/processing.rs

```diff
@@ -0,0 +1,255 @@
+    });
+
+    let receive_from_others = rx.map(Ok).forward(outgoing);
+
+    pin_mut!(broadcast_incoming, receive_from_others);
+    future::select(broadcast_incoming, receive_from_others).await;
+
+    info!("External routing disconnected");
+    disconnect();
+}
+
+fn disconnect() {
+    if CONNECTION.lock().unwrap().is_some() {
+        (*CONNECTION.lock().unwrap()) = None;
+    }
+}
+
+/// Sends a JSON encoded packet to the connected router.
+fn send_packet(p: &Packet) {
+    if let Ok(data) = serde_json::to_string(p) {
+        if let Some(con) = CONNECTION.lock().unwrap().as_ref() {
+            con.tx
+                .unbounded_send(Message::Text(data))
+                .expect("error while sending to tx");
+        }
+    }
+}
+
+/// Takes the RoutingNotification's, encodes them to serializable structs and then sends them
+/// to the external router if one is connected.
+pub fn notify(notification: RoutingNotifcation) {
+    let packet: Packet = match notification {
+        RoutingNotifcation::SendingFailed(bid, cla_sender) => {
+            Packet::SendingFailed(SendingFailed { bid, cla_sender })
+        }
+        RoutingNotifcation::IncomingBundle(bndl) => {
+            Packet::IncomingBundle(IncomingBundle { bndl: bndl.clone() })
+        }
+        RoutingNotifcation::IncomingBundleWithoutPreviousNode(bid, node_name) => {
+            Packet::IncomingBundleWithoutPreviousNode(IncomingBundleWithoutPreviousNode {
+                bid,
+                node_name,
+            })
+        }
+        RoutingNotifcation::EncounteredPeer(eid) => Packet::EncounteredPeer(EncounteredPeer {
+            name: eid.node().unwrap(),
+            eid: eid.clone(),
+            peer: peers_get_for_node(&eid).unwrap(),
+        }),
+        RoutingNotifcation::DroppedPeer(eid) => Packet::DroppedPeer(DroppedPeer {
+            name: eid.node().unwrap(),
+            eid: eid.clone(),
+        }),
+    };
+
+    send_packet(&packet);
+}
+
+fn remove_response_channel(id: &str) {
+    RESPONSES.lock().unwrap().remove(id);
+}
+
+fn create_response_channel(id: &str, tx: oneshot::Sender<Packet>) {
+    RESPONSES.lock().unwrap().insert(id.to_string(), tx);
+}
+
+/// Tries to send a routing requests to the external router and waits for the response.
+/// The wait will be limited to a timeout of 250ms.
+pub async fn sender_for_bundle(bp: &BundlePack) -> (Vec<ClaSenderTask>, bool) {
+    trace!("external sender_for_bundle initiated: {}", bp);
+
+    if CONNECTION.lock().unwrap().is_none() {
+        return (vec![], false);
+    }
+
+    // Register a response channel for the request
+    let (tx, rx) = oneshot::channel();
+    create_response_channel(bp.to_string().as_str(), tx);
+
+    // Send out the SenderForBundle packet
+    let packet: Packet = Packet::SenderForBundle(SenderForBundle {
+        clas: cla_names(),
+        bp: bp.clone(),
+    });
+    send_packet(&packet);
+
+    let res = timeout(time::Duration::from_millis(250), rx).await;
```

Comment:
hardcoded magic numbers should be constants and documented and motivated

Reply:
Changed to a commented constant.